ui/app/src/components/ContentTypeManagement/components/EditTypeView.tsx

```
@@ -481,10 +837,20 @@ function updateTypeFromDataSourceUpdate(
-	updatedType.dataSources[index] = selectedDataSource;
+	const index = updatedType.dataSources.findIndex((item) => {
+		if (selectedDataSource.id) {
+			return item.id === selectedDataSource.id;
+		} else {
+			return (item as NewDataSource).NEW;
+		}
+	});
+
+	const nextDataSource = { ...selectedDataSource };
+	// When updating a new data source, we need to exclude NEW prop from the new datasource content
+	delete (nextDataSource as NewDataSource).NEW;
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)